src/lockscreen.js

```diff
@@ -23,21 +26,42 @@ var Lockscreen = GObject.registerClass({
         super._init(props);
 
         this._overlay = new Gtk.EventBox({
+            aboveChild: true,
             expand: true,
-            visible: true,
         });
-        this._overlay.get_style_context().add_class('lockscreen');
-        this._updateUI();
-
+        this.add_overlay(this._overlay);
         this._overlay.connect('button-release-event', () => {
             this.emit('overlay-clicked');
             return Gdk.EVENT_PROPAGATE;
         });
 
+        this._css_provider = new Gtk.CssProvider();
+        this._style = this._overlay.get_style_context();
+        this._style.add_provider(
+            this._css_provider,
+            Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION
+        );
+        this._playbin = new Playbin({ visible: true });
```

Comment:
Keep in mind that the lock-screen image or animation, should _only_ be shown if the lock screen has never been unlocked before.